draft-yoshino-wish-00.txt

```diff
@@ -321,6 +257,70 @@ Internet-DrafWiSH: A General Purpose Message Framing over B October 2016
    needed.  What contents are exchanged and in what encoding they are
    exchanged over WiSH are to be defined by the application layer.
 
+6.  Using WiSH over HTTP
+
+   Standard HTTP (REST) semantics should be followed, especially the
+   choice of the HTTP method.  Some HTTP semantics may not be
+   applicable, e.g.  Cache-Control, when the body is streamed.  However,
+   such limitation is not specific to WiSH.
+
+7.  WebSocket Compatibility
+
+   In this section, we discuss how to bridge WiSH and WebSocket (API and
+   protocol).
+
+7.1.  Subprotocol Negotiation
+
+   When layered over HTTP, WiSH optionally negotiates the subprotocol,
+   i.e. the protocol layered over WiSH, by using the Accept header and
```

Comment:
when layered over http, the client and server may choose to negotiate a subprotocol (in the WebSocket term) by using the standard HTTP Accept and Content-Type headers.

IN order to be compatible with RFC6455, the client may list the offered subprotocols as follows:

The server chooses one subprotocol from the offered ones and notifies the chosen subprotocol with the content-type header as follows:

The client should not start to stream any data (with the request body) before the client receives all the response headers from the server, which concludes the negotiation process.